anomaly/json_trees_file_loader.cpp

```diff
@@ -0,0 +1,108 @@
+#include "json_trees_file_loader.hpp"
+
+namespace HawkTracer
+{
+namespace anomaly
+{
+
+bool JsonTreesFileLoader::init(const std::string& file_name)
+{
+    _file.open(file_name);
+    if (_file.is_open())
+    {
+        _parse_file();
+    }
+    return _file.is_open();
+}
+
+std::vector<std::pair<std::shared_ptr<CallGraph::TreeNode>, int>> JsonTreesFileLoader::get_trees()
+{
+    return _trees;
+}
+
+template<typename T, typename json_T>
+static void _try_get_value(const jsonxx::Object& json_obj, const std::string& field_name, T& field)
+{
+    if (json_obj.has<json_T>(field_name))
+    {
+        field = json_obj.get<json_T>(field_name);
+    }
+}
+
+void JsonTreesFileLoader::_parse_file()
+{
+    _json_obj.parse(_file);
+    jsonxx::Array trees;
+    _try_get_value<jsonxx::Array, jsonxx::Array>(_json_obj, "trees", trees);
+    for (const auto& tree : trees.values())
+    {
+        _parse_tree(*tree);
+    }
+}
+
+void JsonTreesFileLoader::_parse_tree(const jsonxx::Value& tree)
+{
+    for (const auto& node : tree.array_value_->values())
+    {
+        _parse_node_data(*node);
+    } 
+    for (const auto& node : tree.array_value_->values())
+    {
+        _parse_node_edges(*node);
+    }   
+}
+
+void JsonTreesFileLoader::_parse_node_data(const jsonxx::Value& node)
+{
+    int id;
+    int cnt_calls;
+    int last_start_ts;
+    int last_stop_ts;
+    int total_dur;
+    int total_children_dur;
+    std::string label;
+
+    _try_get_value<std::string, jsonxx::String>(*node.object_value_, "label", label);
+    _try_get_value<int, jsonxx::Number>(*node.object_value_, "id", id);
```

Comment:
How do you know if the value was read correctly? Is it ok to not have `id` or `label`?